akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -658,6 +681,7 @@ private[akka] trait RememberingShard {
  *
  * @see [[ClusterSharding$ ClusterSharding extension]]
  */
+@deprecated("Use `ddata` mode, persistence mode is deprecated.", "2.6.0")
```

Comment:
No need to deprecate the internal API, those we don't promise any compatibility across versions for anyway.